entry_types/paged/app/assets/javascripts/pageflow_paged/server_rendering.js

```diff
@@ -0,0 +1,9 @@
+// Pageflow Paged requires an older version of React than the one that
+// comes bundled with react-rails. The React builds from the
+// originally used version of react-rails have been copied to
+// paged/vendor/assets/javascripts/<env>/pageflow_paged/.
+
+//= require pageflow_paged/react-server
```

Comment:
I agree that this is confusing. I'm mostly unsatisfied, though, with the fact that the vendored React script is called `pageflow_paged/react-server`. It needs to be prefixed with `pageflow_paged` to prevent name collisions, but it looks too much like something that is Pageflow code. So instead of renaming `pageflow_paged/dist/react-server`, I'd consider placing the React script in `vendor/assets/javascripts/<env>/pageflow_paged/vendor/react-server.js`. The require directive above would then look like:

```
//= require pageflow_paged/vendor/react-server
```

That way the file name itself is more redundant, but references look less confusing.

Reply:
In this case, it continues to look to me like there's a vendored file that we somehow massage for distribution. At least, this is what I would probably think if I saw these files for the first time.

Still, I am relatively new to the details of preparing JS code for distribution, so if you told me that to somebody who has dealt with this for a while this naming would be clear, I'd believe you

Reply:
I'd argue that that's what it is: A vendored version of react that is here so it can be distributed along with Pageflow Paged.

We could additionally rename `pageflow_paged/dist/react-server` and  `pageflow_paged/dist/react-client` to `pageflow_paged/dist/pageflow-paged-react-server` and `pageflow_paged/dist/pageflow-paged-react-client` to make the file names match the name of the package they were generated from (i.e. `entry_types/paged/packages/pageflow-paged-react`). But I'd argue that these files already have `pageflow_paged` in their (full) name. 

Moreover, there are also `pageflow_paged/dist/editor.js` and `pageflow_paged/dist/frontend.js` which, by that logic, would need to be called  `pageflow_paged/dist/pageflow-paged-editor.js` and `pageflow_paged/dist/pageflow-paged-frontend.js` (since they are generated from `entry_types/paged/packages/pageflow-paged`).

I feel like this is a slippery slope where we now need to include "pageflow paged" twice in names of things that belong to Pageflow Paged to differentiate them from something that also needs to have "pageflow paged" in the name, but does _not_ really belong to Pageflow Paged (i.e. the vendored React build).

Reply:
Ah, let me clarify: What confused me initially was that we require two files that are named `react-server.js`. This looked to me like an oversight, and it seemed likely we only want to include one of them.

This is what I mean when I write about massaging a vendored file for distribution: The file in `dist` looks like the massaged version of the file in `vendor`, going just by their file names and require paths.

Maybe there isn't a good solution in the file names and directories here, then. Perhaps an inline comment in `components.js` could clarify that despite the identical file names, we're not dealing with an oversight here

Reply:
What could the comment say?

I'd argue that, if you encounter two places each with one of the following directives
```
//= require react-server
//= require pageflow_paged/react-server
```
it would be reasonably safe to assume that the first one is a build of React (for the server side) and the second is Pageflow Paged's React based functionality (for the server side). This is basically what we had before.

If now those turn into 
```
//= require pageflow_paged/vendor/react-server
//= require pageflow_paged/react-server
```
I'd argue it's clear enough that the one in the vendor directory is the React build.

Reply:
Yes, sure, this would be one of many possible safe-looking assumptions. Without searching through the code and knowing much about the context, though, I might also assume that we have two vendored `react-server` versions, one of them for Paged. Or that we are requiring one of them by accident, because possibly our `dist` version is created so as to include the other file's code. Or whatever else my imagination comes up with in the moment, on seeing two files that have the same name.

In an inline comment, I'd acknowledge that for good reasons, we have two files that are named `react-server.js` but that they do indeed fulfill different functions -- maybe also mention what they are there for -- and that this is not an oversight. If you told me that this approach wouldn't scale well, I'd take that point.

My caveat still holds: It is possible that my confusion is singular and contingent on unusual circumstances, in which case there'd be nothing to change here. Especially if the files are placed and named according to common conventions it might even be harmful to change this situation

Reply:
I see your point. I now made the change to call the file `pageflow_paged/vendor/react-sever` which I think is an improvement. I'd like to leave it at that for now. `components.js` does include a comment stating:

https://github.com/codevise/pageflow/blob/1cb852def46a8eee6bfe3f456e2ff14496cac5c8/entry_types/paged/app/assets/javascripts/pageflow_paged/components.js#L1-L2

In this file we have the comment about the vendored React version. So there is at least _something_ regarding purpose.

Reply:
Yes. Still, the comment about `pageflow-paged-react` was too abstract for me to be able to tell what's going on without looking into it a bit.

We'll surely both remember this discussion in case another dev reports confusion about this. Until then, I'd say it's good enough for me